src/collective/fingerpointing/subscribers/genericsetup_logger.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+"""Event subscriber handler for IProfileImportedEvent."""
+from collective.fingerpointing.config import AUDIT_MESSAGE
+from collective.fingerpointing.interfaces import IFingerPointingSettings
+from collective.fingerpointing.logger import log_info
+from collective.fingerpointing.utils import get_request_information
+from plone import api
+
+
+def profile_imports_logger(event):
+    """Log genericsetup profile imports."""
+    record = IFingerPointingSettings.__identifier__ + '.audit_profile_imports'
+    audit_profile_imports = api.portal.get_registry_record(record, default=False)  # noqa: E501
+    if not audit_profile_imports:
+        return
+
+    try:
+        user, ip = get_request_information()
+    except AttributeError:
+        # we're running tests and there's no request available
+        user, ip = '-', '-'
```

Comment:
well, this can be mocked in tests.

Reply:
yes, I tried but there's no request at that point and I don't know how to create one.

any hints?

Reply:
in fact I have a very weird request object:

```pypdb
(Pdb) print self.layer['request']
<h3>form</h3><table></table><h3>cookies</h3><table></table><h3>lazy items</h3><table></table><h3>other</h3><table><tr valign="top" align="left"><th>LANGUAGE</th><td>'en'</td></tr><tr valign="top" align="left"><th>URL</th><td>'http://nohost'</td></tr><tr valign="top" align="left"><th>SERVER_URL</th><td>'http://nohost'</td></tr><tr valign="top" align="left"><th>LANGUAGE_TOOL</th><td>&lt;plone.i18n.utility.LanguageBinding object at 0x7f47cf9028d0&gt;</td></tr><tr valign="top" align="left"><th>method</th><td>'GET'</td></tr><tr valign="top" align="left"><th>ACTUAL_URL</th><td>'http://nohost'</td></tr><tr valign="top" align="left"><th>URL0</th><td>http://nohost/noobject</td></tr><tr valign="top" align="left"><th>URL1</th><td>http://nohost</td></tr><tr valign="top" align="left"><th>BASE0</th><td>http://nohost</td></tr><tr valign="top" align="left"><th>BASE1</th><td>http://nohost</td></tr><tr valign="top" align="left"><th>BASE2</th><td>http://nohost/noobject</td></tr></table><h3>environ</h3><table><tr valign="top" align="left"><th>REQUEST_METHOD</th><td>'GET'</td></tr><tr valign="top" align="left"><th>SERVER_NAME</th><td>'nohost'</td></tr><tr valign="top" align="left"><th>SERVER_PORT</th><td>'80'</td></tr><tr valign="top" align="left"><th>HTTP_X_THEME_ENABLED</th><td>True</td></tr></table>
```

Reply:
the problem is `getResquest()` fails on this particular tests and returns `None`.